broadinstitute/single_cell_portal_core
# Log error modal message to Mixpanel (SCP-3053)

Currently, server- and client-side errors are logged to Mixpanel when they are caught.  However, there is a large number of potential errors that are averted before a potential exception is thrown (e.g. trying to pull up a study by an accession that does not exist) that are simply shown to the user as an "alert modal", and are not reported anywhere else.  This update adds a `log` call from the `metrics-api.js` library to report the content of the error text to Mixpanel.  This will help identify any usability issues that the portal has been handling silently.

This update also extends server-side error reporting to Mixpanel of explicitly handled exceptions.  Previously only uncaught exceptions were logged to Mixpanel via `MetricsService.report_error()`.

To test:

1. Pull `jb-mixpanel-error-modal` locally and boot development instance
2. When home page loads, navigate to https://localhost/single_cell/study/SCP0 (or https://localhost:3000/single_cell/study/SCP0 if running non-Dockerized)
3. Once modal renders with `Study not found` message, open Dev Tools > Network and observe Mixpanel `POST` event with `event: "error-modal"` title
4. Alternatively, go to https://mixpanel.com/report/2085496/view/19055/dashboards and create a new Insights report using the `error-modal` event

This PR satisfies SCP-3053.
